Terminal.Gui/Drawing/LineCanvas.cs

```diff
@@ -249,99 +277,118 @@ private abstract class IntersectionRuneResolver {
 			readonly Rune doubleH;
 			readonly Rune doubleV;
 			readonly Rune doubleBoth;
+			readonly Rune thickH;
+			readonly Rune thickV;
+			readonly Rune thickBoth;
 			readonly Rune normal;
 
-			public IntersectionRuneResolver (Rune round, Rune doubleH, Rune doubleV, Rune doubleBoth, Rune normal)
+			public IntersectionRuneResolver (Rune round, Rune doubleH, Rune doubleV, Rune doubleBoth, Rune thickH, Rune thickV, Rune thickBoth, Rune normal)
 			{
 				this.round = round;
 				this.doubleH = doubleH;
 				this.doubleV = doubleV;
 				this.doubleBoth = doubleBoth;
+				this.thickH = thickH;
+				this.thickV = thickV;
+				this.thickBoth = thickBoth;
 				this.normal = normal;
 			}
 
 			public Rune? GetRuneForIntersects (ConsoleDriver driver, IntersectionDefinition [] intersects)
 			{
-				var useRounded = intersects.Any (i => i.Line.Style == LineStyle.Rounded && i.Line.Length != 0);
+				var useRounded = intersects.Any (i => i.Line.Length != 0 && (
+					i.Line.Style == LineStyle.Rounded || i.Line.Style == LineStyle.RoundedDashed || i.Line.Style == LineStyle.RoundedDotted));
+
+				// Note that there aren't any glyphs for intersections of double lines with thick lines
 
 				bool doubleHorizontal = intersects.Any (l => l.Line.Orientation == Orientation.Horizontal && l.Line.Style == LineStyle.Double);
 				bool doubleVertical = intersects.Any (l => l.Line.Orientation == Orientation.Vertical && l.Line.Style == LineStyle.Double);
 
+				bool thickHorizontal = intersects.Any (l => l.Line.Orientation == Orientation.Horizontal && (
+					l.Line.Style == LineStyle.Thick || l.Line.Style == LineStyle.ThickDashed || l.Line.Style == LineStyle.ThickDotted));
+				bool thickVertical = intersects.Any (l => l.Line.Orientation == Orientation.Vertical && (
+					l.Line.Style == LineStyle.Thick || l.Line.Style == LineStyle.ThickDashed || l.Line.Style == LineStyle.ThickDotted));
 
 				if (doubleHorizontal) {
 					return doubleVertical ? doubleBoth : doubleH;
 				}
-
 				if (doubleVertical) {
 					return doubleV;
 				}
 
+				if (thickHorizontal) {
```

Comment:
I think there's  bug in here somewhere:

<img width="435" alt="image" src="https://user-images.githubusercontent.com/585482/232041457-d3266fb6-cadf-4136-bd6b-a2009f4c317f.png">

Reply:
And it looks like ThickDotted has same issue (horizontal is not getting the right glyph)

Reply:
> The ~padding~ border in this image isn't rendering well.
> 
> **Edit:** Sorry, my bad. Is the border limits that isn't rendering well. The border frame is correct but not the LineCanvas rendering.

This is intentional for now; I want clipping turned off until we are sure everyone is drawing in the right places.

Reply:
> This is intentional for now; I want clipping turned off until we are sure everyone is drawing in the right places.

Yes I know, but I'm talking about this:

![imagem](https://user-images.githubusercontent.com/13117724/232103611-0d1f07eb-ce47-4b2d-bc10-8523adf37fd7.png)

The border `LineCanvas` should be inside the border frame and it isn't.

Reply:
I see. Yep. That's a bug.